stackstate_checks_base/stackstate_checks/base/utils/http_helper.py

```diff
@@ -0,0 +1,1222 @@
+import json
+import requests_mock
+import logging
+from enum import Enum
+from schematics.models import Model
+from schematics.types import BaseType, StringType, IntType, DictType, BooleanType
+from schematics.exceptions import DataError
+from requests import Session, Request
+from requests.auth import HTTPBasicAuth
+from requests.adapters import HTTPAdapter
+from requests.packages.urllib3.util.retry import Retry
+
+try:
+    import urlparse
+except ImportError:
+    import urllib.parse as urlparse
+
+
+class HTTPBasicAuthentication(Model):
+    username = StringType(required=True)
+    password = StringType(required=True)
+
+
+class HTTPAuthenticationType(Enum):
+    BasicAuth = HTTPBasicAuthentication
+
+
+class HTTPMethod(Enum):
+    GET = 'GET'
+    POST = 'POST'
+    PUT = 'PUT'
+    PATCH = 'PATCH'
+    DELETE = 'DELETE'
+
+
+class HTTPRequestType(Enum):
+    JSON = dict
+
+
+class HTTPResponseType(Enum):
+    PLAIN = str
+    JSON = dict
+
+
+class HTTPHelperModel(Model):
```

Comment:
It's good that you didn't get tempted to pass this model down to all classes. That would've saved you some code, but would have made entangled all your code and would've made it hard to disentangle at some later stage when you want to use/test things in isolation 👍